libs/ledger-live-common/src/families/cosmos/api/Cosmos.unit.test.ts

```diff
@@ -0,0 +1,797 @@
+      } as AxiosResponse);
+
+      const account = await cosmosApi.getAccount("addr");
+      expect(account).toEqual({
+        accountNumber: 1,
+        sequence: 0,
+        pubKey: "k",
+        pubKeyType: "type",
+      });
+
+      mockedNetwork.mockResolvedValue({
+        data: {
+          account: {
+            "@type": "/ethermint.types.v1.EthAccount",
+            account_number: 1,
+            sequence: 0,
+            pub_key: { key: "k", "@type": "type" },
+
+            base_account: {
+              address: "address",
+              pub_key: { key: "k2", "@type": "type2" },
+              account_number: 2,
+              sequence: 3,
+            },
+            code_hash: "codeHash",
+          },
+        },
+      } as AxiosResponse);
+
+      const ethermintAccount = await cosmosApi.getAccount("addr");
+      expect(ethermintAccount).toEqual({
+        accountNumber: 2,
+        sequence: 3,
+        pubKey: "k2",
+        pubKeyType: "type2",
+      });
+    });
+
+    it("should return default sequence value if network fails", async () => {
+      mockedNetwork.mockImplementation(() => {
+        throw new Error();
+      });
+      const account = await cosmosApi.getAccount("addr");
+      expect(account.sequence).toEqual(0);
+    });
+
+    it("should return default pubkeytype value if network fails", async () => {
+      mockedNetwork.mockImplementation(() => {
+        throw new Error();
+      });
+      const account = await cosmosApi.getAccount("addr");
+      expect(account.pubKeyType).toEqual(cryptoFactory("cosmos").defaultPubKeyType);
+    });
+  });
+
+  describe("simulate", () => {
+    it("should return gas used when the network call returns gas used", async () => {
+      mockedNetwork.mockResolvedValue({
+        data: {
+          gas_info: {
+            gas_used: 42000,
+          },
+        },
+      } as AxiosResponse);
+      const gas = await cosmosApi.simulate([]);
+      expect(gas).toEqual(new BigNumber(42000));
+    });
+
+    it("should throw an error when the network call does not return gas used", async () => {
+      mockedNetwork.mockResolvedValue({
+        data: { gas_info: {} },
+      } as AxiosResponse);
+      await expect(cosmosApi.simulate([])).rejects.toThrowError();
+    });
+
+    it("should throw an error when the network call fails", async () => {
+      mockedNetwork.mockImplementation(() => {
+        throw new Error();
+      });
+      await expect(cosmosApi.simulate([])).rejects.toThrowError();
+    });
+  });
+
+  describe("getTransactions", () => {
+    it("should return an empty array when network call fails", async () => {
+      // @ts-expect-error method is mocked
+      network.mockImplementation(() => {
+        throw new Error();
+      });
+      const txs = await cosmosApi.getTransactions("address", 50);
+      expect(txs.length).toEqual(0);
+    });
+
+    // it("should not run into an infinite loop if every call return the same thing", async () => {
```

Comment:
Let's remove it if you think it's not useful anymore

Reply:
It shouldn't be necessary as we're relying on the total + offset to know when to stop. 
To be honest I really don't like the do/while method, I wanted to propose a recursive version with a nicer way to stop in case of a catastrophic failure, but I'll keep that as an improvement. 
For now at least, there is no visible reason for it to enter an infinite loop imo